File: 560-SubarraySumEqualsK/560-SubarraySumEqualsK.java
// Last updated: 26/12/2025, 16:00:46
class Solution {
    public int subarraySum(int[] nums, int k) {
        if (nums.length == 0) return 0;
        
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(0,1);
        int sum=0, result=0;
        
        for(int i=0; i<nums.length; i++){
            sum+=nums[i];
            result += map.getOrDefault(sum-k,0);
            map.put(sum,map.getOrDefault(sum,0)+1);
        }
        
        return result;
    }
}

File: 560-SubarraySumEqualsK/560-SubarraySumEqualsK.js
// Last updated: 26/12/2025, 16:00:42
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function(nums, k) {
    let map = new Map()
    map.set(0,1)
    let sum = 0
    let result = 0

    for(let i = 0; i< nums.length; i++){
        sum += nums[i]
       result += (map.get(sum-k) || 0)
       map.set(sum, (map.get(sum) || 0)+1)

    }

    return result 
};


